atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/KvsMigrationCommand.java

```diff
@@ -192,4 +247,26 @@ private KeyValueServiceMigrator getMigrator(AtlasDbServices fromServices, AtlasD
                 .batchSize(batchSize)
                 .build());
     }
+
+    private static <K> Optional<K> loadFromFileOrInline(
+            @Nullable File configFile, String fileConfigRoot, @Nullable String inline, Class<K> clazz) {
+        return loadFromFile(configFile, fileConfigRoot, clazz).or(() -> Optional.ofNullable(inline)
+                .map(rethrowIoExceptionAsUnchecked(config -> AtlasDbConfigs.loadFromString(config, null, clazz))));
+    }
+
+    private static <K> Optional<K> loadFromFile(@Nullable File configFile, String fileConfigRoot, Class<K> clazz) {
+        return Optional.ofNullable(configFile)
+                .map(rethrowIoExceptionAsUnchecked(file -> AtlasDbConfigs.load(file, fileConfigRoot, clazz)));
+    }
+
+    private static <F, T, K extends IOException> Function<F, T> rethrowIoExceptionAsUnchecked(
```

Comment:
map doesn't support the lambda throwing checked exceptions, so I've a helper function that rethrows IO exceptions to the unchecked equivalent. There is prior art for this in the CLI with Throwables.rewrapAndThrowUncheckedException.

The alternative would be not to use map, but I figured this was cleaner.